src/main/java/model/ModelGraph.java

```diff
@@ -129,6 +129,18 @@ public void deleteEdge(GraphNode n1, GraphNode n2){
         return Optional.ofNullable(edges.get(id));
     }
 
+    public Optional<Vertex> getVertexBetween(Vertex beginning, Vertex end){
+        return this.vertexes
+                .values()
+                .stream()
+                .filter(v -> isVertexBetween(v, beginning, end))
+                .findFirst();
+    }
+
+    private boolean isVertexBetween(Vertex v, Vertex beginning, Vertex end){
```

Comment:
Why we cannot use this API method:
beginning.getEdgeBetween(v.getIndex()) != null ????
Due to this we don't need this part of code from Vertex class:
@Override
    public <T extends Edge> T getEdgeBetween(Node node) {

Reply:
Short answer: it doesn't work.

It outputs null:
![image](https://user-images.githubusercontent.com/15652452/67161831-998ada00-f35e-11e9-9631-7f6bb22b7125.png)

Even though edges from "v1 to h1" and "h1 to v2" exist:
![image](https://user-images.githubusercontent.com/15652452/67161842-a7d8f600-f35e-11e9-875c-4e36606c733c.png)

I suspect that it is because internally it uses AbstractEdge/AbstractNode and hashcodes are not matching in `neighborMap`. Changing from `.getId` to `.getIndex` doesn't fix it